### PR TITLE
Read-only stress test with better diagnostics

### PR DESCRIPTION
**Pull request by jskeet**
(Builds on #2648)

### CODE REVIEW

File: apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ReliableStreamReader.cs [hunk 1]
```diff
@@ -64,6 +64,8 @@ public sealed class ReliableStreamReader : IDisposable
             _request.SessionAsSessionName = _session.SessionName;
         }
 
+        // TODO: Take this in the constructor instead? Or maybe put it in SpannerClientSettings?
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ReliableStreamReader.cs]: Wouldn't the one in SpannerClientSettings be the one that _spannerClient is using? I would use the same as _spannerClient (if as I remember it has one). But OK with the TODO.

Comment (jskeet) [reply apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ReliableStreamReader.cs]: Nope, there isn't one in SpannerClientSettings at the moment.

Comment (amanda-tarafa) [reply apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ReliableStreamReader.cs]: Oh, I meant if `SpannerClientSettings` were to have one, then `SpannerClient` would use that one I suppose, so if here we use the one that `SpannerClient` currently has, we don't care if it came from settings or somewhere else, and it makes sense for everything to use the one in `SpannerClient` anyways, I think? But happy with deferring this.

Comment (jskeet) [reply apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ReliableStreamReader.cs]: Yes, if we created a Logger property in SpannerClientSettings, we could use that in multiple places.

**Pull request merged**